mobile/src/main/java/com/joaquimley/avenging/ui/list/ListFragment.java

```diff
@@ -316,6 +318,12 @@ public boolean onQueryTextChange(String queryText) {
     }
 
     @Override
+    public void onDestroyView() {
+        mCharactersRecycler.setAdapter(null);
```

Comment:
On a closer inspection, the only reference the adapter has to the fragment is the `ListInteractionListener`. Have you tried to add `mListCharacterAdapter.setListInteractionListener(null);` instead and see if this fixes the leak.